modules/InboundEmail/InboundEmail.php

```diff
@@ -5268,39 +5268,42 @@ public function returnNonImportedEmail($msgNo, $uid)
             $email->type = 'inbound';
 
 
-            if (empty($email->date_entered)) {
-
-                // find if string has (UTC) in timezone
-                $pattern = "/\([\w-+\/]+\)/i";
-                $timezoneTextFound = preg_match($pattern, $header[0]->date);
-
-                $dateTimeFormat = 'D, d M Y H:i:s O *';
-                if ($timezoneTextFound === false || $timezoneTextFound === 0) {
-                    $dateTimeFormat = 'D, d M Y H:i:s O';
+            if(empty($email->date_entered)) {
+
+                $possibleFormats = [
+                    \DateTime::RFC2822,
+                    str_replace(['D, '], '', \DateTime::RFC2822), // day-of-week is optional
+                    str_replace([':s'], '', \DateTime::RFC2822), // seconds are optional
+                    str_replace(['D, ', ':s'], '', \DateTime::RFC2822), // day-of-week is optional, seconds are optional
+                    \DateTime::RFC822,
+                    str_replace(['D, '], '', \DateTime::RFC822), // day is optional
```

Comment:
Shouldn't day be 'd' ?

Reply:
@PedroErnst No, this is referring to the leading "day of week" variable [PHP DateTime](http://php.net/manual/en/class.datetime.php). The comment refers to what the RFC calls it. In 2822 they refer to it as "day-of-week"; in 822 they refer to it as "day". 

Both actually refer to what PHP calls 
> D | A textual representation of a day, three letters | Mon through Sun

[Source](http://php.net/manual/en/function.date.php#refsect1-function.date-parameters)